src/js/panels.js

```diff
@@ -600,204 +627,219 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+                // the language that we want to scroll to is below
+                var newIndex = langIndex - that.options.numOfLangPerPage + 1;
+                that.applier.change("displayLangIndex", newIndex);
+            }
         }
     };
 
-    gpii.firstDiscovery.panel.lang.overrideDefaultScroll = function (that) {
-        that.locate("controlsDiv").scroll(function () {
-            gpii.firstDiscovery.panel.lang.scrollLangIntoView(that);
-        });
+    gpii.firstDiscovery.panel.lang.displayActiveLang = function (that, lang) {
+        var langIndex = that.options.controlValues.lang.indexOf(lang);
+        if (langIndex < that.options.numOfLangPerPage) {
+            // if our active language is on the first page, display
+            // from the start of the list
+            langIndex = 0;
+        }
+        that.applier.change("displayLangIndex", langIndex);
     };
 
-    gpii.firstDiscovery.panel.lang.stopArrowBrowseOnEdgeButtons = function (button, keyCodes) {
-        $(button).keydown(function (e) {
-            if (keyCodes.indexOf(e.which) !== -1) {
-                e.preventDefault();
-                return false;
-            }
-        });
+    gpii.firstDiscovery.panel.lang.updateDisplayedLangs = function (that, langIndex) {
+        var buttons = that.locate("langRow");
+        if (buttons.length > 0) {
+            var firstButtonTop = buttons.offset().top;
+            var displayFromButton = $(buttons[langIndex]);
+            var scrollTo = displayFromButton.offset().top - firstButtonTop;
+            $(that.options.selectors.controlsDiv).animate({scrollTop: scrollTo}, 0);
+        }
+        that.events.displayedLangsUpdated.fire();
     };
 
-    // When the focus is on the first language button, prevent the press of up or left arrow keys moving to the last language button;
-    // when the focus is on the last language button, prevent the press of down or right arrow keys moving to the first language button.
-    // TODO: Replace this funciton with fluid.selectable() plugin with noWrap: true when FLUID-5642 (http://issues.fluidproject.org/browse/FLUID-5642)
-    // is fixed.
-    gpii.firstDiscovery.panel.lang.preventWrapWithArrowKeys = function (that) {
-        var langButtons = that.locate("langInput"),
-            firstLangButton = langButtons[0],
-            lastLangButton = langButtons[langButtons.length];
-
-        gpii.firstDiscovery.panel.lang.stopArrowBrowseOnEdgeButtons(firstLangButton, [$.ui.keyCode.UP, $.ui.keyCode.LEFT]);
-        gpii.firstDiscovery.panel.lang.stopArrowBrowseOnEdgeButtons(lastLangButton, [$.ui.keyCode.DOWN, $.ui.keyCode.RIGHT]);
+    gpii.firstDiscovery.panel.lang.refreshDisplayedLangsOnShowPanel = function (that, shownPanelId) {
+        var langPanelId = that.container.attr("id");
+        if (langPanelId === shownPanelId) {
+            // reset back to the top of the list and refresh
+            $(that.options.selectors.controlsDiv).animate({scroolTop: 0}, 0);
```

Comment:
Given both of these durations are 0, wouldn't it be simpler to just issue the .scrollTop change directly?